heareval/predictions/task_predictions.py

```diff
@@ -89,10 +102,11 @@ def validation_step(self, batch, batch_idx):
         x, y = batch
         y_hat = self.predictor.forward_logit(x)
         loss = self.predictor.logit_loss(y_hat, y)
+        y_pr = self.predictor(x)
         # Logging to TensorBoard by default
-        self.log("val_loss", loss)
-        # TODO: Use the actual evaluation criterion?
-        return loss
+        self.log("val_loss", loss, prog_bar=True)
+        for score in self.scores:
+            self.log(f"val_{score}", score(y_pr, y), prog_bar=True)
```

Comment:
Should this still return a loss? I see that you're logging the validation now -- what do we use for early stopping now?

Reply:
I think I get it -- see comment below

Reply:
I can also see how this is awkward for event based metrics :/

Reply:
Yeah. In a later PR I am accumulating at the end of validation, not every validation step. But it's still a pain for event-based metrics.